firebase-config/src/main/java/com/google/firebase/remoteconfig/RemoteConfigComponent.java

```diff
@@ -299,4 +315,26 @@ private static boolean isAbtSupported(FirebaseApp firebaseApp, String namespace)
   private static boolean isPrimaryApp(FirebaseApp firebaseApp) {
     return firebaseApp.getName().equals(FirebaseApp.DEFAULT_APP_NAME);
   }
+
+  private static class GlobalBackgroundListener implements BackgroundDetector.BackgroundStateChangeListener {
+    private static AtomicReference<GlobalBackgroundListener> INSTANCE = new AtomicReference<>();
+
+    private static void ensureBackgroundListenerIsRegistered(Context context) {
+      Application application = (Application) context.getApplicationContext();
+      if (INSTANCE.get() == null) {
+        GlobalBackgroundListener globalBackgroundListener = new GlobalBackgroundListener();
+        if (INSTANCE.compareAndSet(null, globalBackgroundListener)) {
+          BackgroundDetector.initialize(application);
```

Comment:
How do we stop the listener? Will there be an API to add/remove the listener?

Reply:
I can't find a way to remove a listener from the BackgroundDetector but I added a flag in FirebaseRemoteConfig that can enable foreground rekicking or disable it.